diagnostic_aggregator/include/diagnostic_aggregator/analyzer.hpp

```diff
@@ -88,60 +93,66 @@ class Analyzer
   /*!
    *\brief Default constructor, called by pluginlib.
    */
-  Analyzer() {}
+  Analyzer()
+  : clock_(new rclcpp::Clock()) {}
 
   virtual ~Analyzer() {}
 
   /*!
    *\brief Analyzer is initialized with base path and namespace.
    *
-   * The Analyzer initialized with parameters in its given 
+   * The Analyzer initialized with parameters in its given
    * namespace. The "base_path" is common to all analyzers, and needs to be
    * prepended to all DiagnosticStatus names.
    *\param base_path : Common to all analyzers, prepended to all processed names. Starts with "/".
+   *\param breadcrumb : Prefix for parameter getter.
    *\param n : NodeHandle with proper private namespace for analyzer.
    */
-  virtual bool init(const std::string base_path, const ros::NodeHandle &n) = 0;
+  virtual bool init(
+    const std::string &,
+    const std::string &,
+    const rclcpp::Node::SharedPtr) = 0;
```

Comment:
I think it makes more sense to pass in the node ptr in the Analyzer constructor.

Reply:
I think this is passed in `init()` instead of the constructor, as the constructor can't have any arguments to to the pluginlib (the `analyzer_loader` will raise errors, when constructor arguments are introduced).